pkg/newlog/cmd/vector.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	uploadSockVectorSource = "/run/devUpload_source.sock"
+	keepSockVectorSource   = "/run/devKeep_source.sock"
+	uploadSockVectorSink   = "/run/devUpload_sink.sock"
+	keepSockVectorSink     = "/run/devKeep_sink.sock"
+
+	defaultConfigPath   = "/persist/vector/config/vector.yaml.default"
+	candidateConfigPath = "/persist/vector/config/vector.yaml.new"
+)
+
+// listenOnSocketAndWriteToChan - goroutine to listen on unix sockets for incoming log entries
+func listenOnSocketAndWriteToChan(sockPath string, sendToChan chan<- string) {
+	// Create unix socket
+	os.Remove(sockPath) // Remove any existing socket
+	unixAddr, err := net.ResolveUnixAddr("unix", sockPath)
+	if err != nil {
+		log.Fatalf("createIncomingSockListener: ResolveUnixAddr failed: %v", err)
+	}
+	unixListener, err := net.ListenUnix("unix", unixAddr)
+	if err != nil {
+		log.Fatalf("createIncomingSockListener: ListenUnix failed: %v", err)
+	}
+	defer unixListener.Close()
+	defer os.Remove(sockPath)
+
+	// Set permissions on socket
+	if err := os.Chmod(sockPath, 0666); err != nil {
+		log.Fatalf("createIncomingSockListener: chmod socket failed: %v", err)
+	}
+
+	// Handle socket connections
+	for {
+		conn, err := unixListener.Accept()
+		if err != nil {
+			log.Errorf("createIncomingSockListener: upload accept failed: %v", err)
+			continue
+		}
+		go handleIncomingConnection(conn, sendToChan)
+	}
+}
+
+// handleIncomingConnection processes incoming log entries from unix socket connections
+func handleIncomingConnection(conn net.Conn, sendToChan chan<- string) {
+	defer conn.Close()
+
+	scanner := bufio.NewScanner(conn)
+	for scanner.Scan() {
+		logline := scanner.Text()
+		if logline == "" {
+			continue
+		}
+
+		// add newline character to the end of the logline
+		if !strings.HasSuffix(logline, "\n") {
+			logline += "\n"
+		}
+
+		sendToChan <- logline
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Errorf("handleIncomingConnection: scanner error for socket: %v", err)
+	}
+}
+
+func writeVectorConfig(text []byte) error {
+	// Create the directory if it doesn't exist
+	if err := os.MkdirAll(filepath.Dir(candidateConfigPath), 0755); err != nil {
+		return fmt.Errorf("failed to create config directory: %w", err)
+	}
+
+	if err := os.WriteFile(candidateConfigPath, text, 0644); err != nil {
```

Comment:
We need at least a WriteRename here to deal with power outages without causing truncated files.
Might also need the ability to fall back to the previous (.bak) fiile from before the last WriteRename.

Reply:
Done